activesupport/lib/active_support/cache/memory_store.rb

```diff
@@ -210,7 +210,10 @@ def read_entry(key, **options)
         def write_entry(key, entry, **options)
           payload = serialize_entry(entry, **options)
           synchronize do
-            return false if options[:unless_exist] && exist?(key)
+            if options[:unless_exist]
+              entry = read_entry(key, **options)
+              return false if entry && !entry.expired? && !entry.mismatched?(normalize_version(key, options))
+            end
```

Comment:
So the problem with using `exist?(key)` is that, by this point, `key` is already namespaced and then [`exist?` _also_ applies the namespace from the base options](https://github.com/rails/rails/blob/139c5678aa5d9a4209ca06919abea7a9da449b36/activesupport/lib/active_support/cache.rb#L711), correct?

In that case, we could just specify `namespace: nil` instead of duplicating `exist?`'s logic:

```suggestion
            return false if options[:unless_exist] && exist?(key, namespace: nil)
```

Reply:
calling exist could bring other side-effects too -- eg. it calls merge_options (again - it's already been done further up the current call stack)